model_analyzer/result/result_comparator.py

```diff
@@ -0,0 +1,202 @@
+            If None, then equal weighting (normal priority comparison)
+        """
+
+        # Index the non gpu metric types
+        self._non_gpu_type_to_idx = {
+            non_gpu_metric_types[i]: i
+            for i in range(len(non_gpu_metric_types))
+        }
+
+        # Index the gpu metric types
+        self._gpu_type_to_idx = {
+            gpu_metric_types[i]: i
+            for i in range(len(gpu_metric_types))
+        }
+
+        self._metric_priorities = metric_priorities
+        self._comparison_threshold_factor = (comparison_threshold_percent *
+                                             1.0) / 100
+
+        # TODO implement a weighted comparison
+        self._weights = weights
+
+    def compare_results(self, result1, result2):
+        """
+        Compares two results using priorities 
+        specified in model analyzer config
+
+        Parameters
+        ----------
+        result1 : RunResult
+            first result to be compared
+        result2 : RunResult
+            second result to be compared
+        
+        Returns
+        -------
+        int 
+            0 
+                if the results are determined
+                to be the same within a threshold
+            1
+                if result1 > result2
+            -1
+                if result1 < result2
+        """
+
+        # For now create an average measurment
+        agg_measurement1 = self._aggregate_measurements(
+            result=result1, aggregation_func=average_list)
+        agg_measurement2 = self._aggregate_measurements(
+            result=result2, aggregation_func=average_list)
+
+        return self.compare_measurements(measurement1=agg_measurement1,
+                                         measurement2=agg_measurement2)
+
+    def compare_measurements(self, measurement1, measurement2):
+        """
+        Compares individual meausurements retrieved from perf runs
+        based on priorities specified by model analyzer config
+
+        Parameters
+        ----------
+        measurement1 : Measurement
+            first set of (gpu_measurements, non_gpu_measurements) to
+            be compared
+        measurement2 : Measurement
+            first set of (gpu_measurements, non_gpu_measurements) to
+            be compared
+
+        Returns
+        -------
+        int 
+            0 
+                if the results are determined
+                to be the same within a threshold
+            1
+                if measurement1 > measurement2
+            -1
+                if measurement1 < measurement2
+        """
+
+        gpu_data1 = measurement1.gpu_data()
+        gpu_data2 = measurement2.gpu_data()
+
+        gpu_rows1 = []
+        gpu_rows2 = []
+
+        for _, gpu_row in gpu_data1.items():
+            gpu_rows1.append(gpu_row)
+        for _, gpu_row in gpu_data2.items():
+            gpu_rows2.append(gpu_row)
+
+        avg_gpu_data1 = average_list(gpu_rows1)
```

Comment:
If I understood correctly, the reason that we need to separate gpu_data and non_gpu data is that gpu_data is a list of measurements while the perf_analyzer result is a single number. I think it would be great if we can aggregate all the GPU (or group of metrics) into a single number for each metric being monitored. So that in the measurement class, we have a list of metrics and a single number representing the result for that specific metric. Then, we can treat all the metrics uniformly without having to separate them.

Reply:
You're saying we don't need the per gpu data? I explicitly average over GPU ID in the result comparator, but do you mean we should do that earlier/elsewhere in the process?

Reply:
I see. I thought the reason that this is split out is because of the GPU metrics being a list of measurements per GPU not because we may have multiple GPUs.